lms/tasks/roster.py

```diff
@@ -72,6 +81,15 @@ def schedule_fetching_course_rosters() -> None:
             )
             for lms_course_id in request.db.scalars(query).all():
                 fetch_course_roster.delay(lms_course_id=lms_course_id)
+                # Record that the roster fetching has been scheduled
+                # We set the expiration date to ROSTER_REFRESH_WINDOW so we'll try again after that period
+                request.db.add(
+                    TaskDone(
+                        key=f"roster::course::{lms_course_id}",
+                        data=None,
+                        expires_at=datetime.now() + ROSTER_REFRESH_WINDOW,
```

Comment:
expired TaskDone are cleaned from the DB by a celery task.

Reply:
Shouldn't we create the `task_done` entry only after the roster fetching has finished successfully or failed?

It doesn't feel super intuitive that we are creating something called "task_done" when the task is not yet done, because at this point it is only being scheduled.

Unless I'm missing something.

Reply:
> Shouldn't we create the task_done entry only after the roster fetching has finished successfully or failed?

That could work as well but we'd miss the case where the roster schedule happens faster than the fetching retries, ie we schedule new fetches every 5 minutes but we do retries that span longer than that in total.


> It doesn't feel super intuitive that we are creating something called "task_done" when the task is not yet done

I do agree with that, I could call the TaskDone key `roster::course::scheduled` or similar, what's is done is not the fetching but the scheduling.

Reply:
> I do agree with that, I could call the TaskDone key `roster::course::scheduled` or similar, what's is done is not the fetching but the scheduling.

I think that should do.

Reply:
@acelaya  Done :+1: